airflow/models/xcom.py

```diff
@@ -376,4 +505,7 @@ def resolve_xcom_backend():
     return BaseXCom
 
 
-XCom = resolve_xcom_backend()
+if TYPE_CHECKING:
+    XCom = BaseXCom  # Hack to avoid Mypy "Variable 'XCom' is not valid as a type".
+else:
+    XCom = resolve_xcom_backend()
```

Comment:
I think this might be causing docs to fail too, as it's somehow finding two places for all the methods on XCom.

(I'm guessing as to cause, doc build is failing though)

Reply:
I think this is fine, the cause is actually the overloads:

```
WARNING: duplicate object description of airflow.models.xcom.BaseXCom.clear, other instance in _api/airflow/models/xcom/index, use :noindex: for one of them

File path: apache-airflow/_api/airflow/models/xcom/index.rst (215)

 210 |    .. classmethod:: clear(cls, execution_date: pendulum.DateTime, dag_id: str, task_id: str, session: Optional[Session] = None)
 211 | 
 212 | 
 213 | 
 214 |    
>215 |    .. classmethod:: clear(cls, execution_date: Optional = None, dag_id: Optional = None, task_id: Optional = None, *, run_id: Optional = None, session: Session)
 216 | 
 217 |       Clears all XCom data from the database for the task instance
 218 | 
 219 |       ``run_id`` and ``execution_date`` are mutually exclusive.
```

I think generated by sphinx-autoapi. I’d expect it to support this _years_ old feature; it’s very disappoiting.

Reply:
Well actually it does support it since last year, but we’re pinning an extremely old version.

Reply:
Every so often I've tried upgrading sphinx-autoapi, but last _n_ times I tried there was a bug that meant it didn't build docs on a more recent version :(

Reply:
Why are we using autoapi in the first place? Do we have any non-Python APIs that can’t be documented with sphinx.ext.autodoc?

Reply:
I think this is question mostly to @mik-laj who knows more of the context. But I share the same sentiment and experience as @ashb - any time I tried I backtracked so far. And yea - we are using extremely old version.